### PR TITLE
Multiple trials

### PR DESCRIPTION
**Pull request by corajr**
Refactored the graphing classes and added an option to run multiple trials and take the average.

### CODE REVIEW

File: lib/graph/function/comparison.rb [hunk 1]
```diff
@@ -12,26 +12,32 @@ def initialize(generator)
...
-        methods.each do |m|
-          self.class.send(:define_method, m.name, proc(&m))
-        end
-
         Gnuplot.open do |gp|
           Gnuplot::Plot.new(gp) do |plot|
 
-            plot.title  "#{title = methods.map {|m| camel_title(m.name) }.join(', ') }"
-						set_up(plot)
+            title = case methods.size
```
Comment (mooreniemi) [inline lib/graph/function/comparison.rb]: perhaps clump this case stmt out into a function?

File: lib/graph/function/comparison.rb [hunk 2]
```diff
@@ -12,26 +12,32 @@ def initialize(generator)
...
             pb = ProgressBar.create(title: title, total: x.size)
 
             methods.each do |m|
               pb.reset
               y = x.collect do |v|
                 pb.increment
                 data = data_generator.call(v)
-                # FIXME can i get ride of the cost of `send`?
-                Benchmark.measure { self.send(m.name, data) }.real
+                (1..trials).collect do |_|
```
Comment (mooreniemi) [inline lib/graph/function/comparison.rb]: nice

File: lib/graph/function/ints_comparison.rb [hunk 3]
```diff
@@ -1,48 +1,13 @@
...
-            pb.reset
-
-            z = x.collect do |v|
-              pb.increment
-              array = (0..v - 1).to_a.shuffle
-              Benchmark.measure { b(array) }.real
-            end
+    class IntsComparison < Comparison
+      def initialize
+        @data_generator = Proc.new {|v| (0..v - 1).to_a.shuffle }
```
Comment (mooreniemi) [inline lib/graph/function/ints_comparison.rb]: now that i think about it should prob have (-v to v - 1) to really cover Integer properly, wdyt?

File: lib/graph/function/comparison.rb (hunk 1 above)
Comment (corajr) [reply lib/graph/function/comparison.rb]: Good thought, moved to a private method in f00a732

File: lib/graph/function/ints_comparison.rb (hunk 3 above)
Comment (corajr) [reply lib/graph/function/ints_comparison.rb]: To be the same size it'd have to be like (-v/2 .. v/2) but I think there's not necessarily a need here -- if negatives are needed one could just write a generator.

**Comment by mooreniemi** [comment]
i added https://github.com/mooreniemi/graph-function/issues/7

**Review by mooreniemi — APPROVED** [review]

**Pull request merged**